src/sentry/api/endpoints/source_map_debug_blue_thunder_edition.py

```diff
@@ -238,34 +239,39 @@ def get(self, request: Request, project: Project, event_id: str) -> Response:
 
 
 def get_source_file_data(abs_path, project, release, event):
-    filenme_choices = ReleaseFile.normalize(abs_path)
+    matching_source_file_names = ReleaseFile.normalize(abs_path)
+    matching_source_map_name = None
+    source_map_reference = None
+    source_file_lookup_result = "unsuccessful"
+    source_map_lookup_result = "unsuccessful"
 
-    path_data = {
-        "abs_path": abs_path,
-        "matching_source_file_names": filenme_choices,
-        "matching_source_map_name": None,
-        "source_map_reference": None,
-        "source_file_lookup_result": "unsuccessful",
-        "source_map_lookup_result": "unsuccessful",
-    }
+    found_source_file_name = None
 
-    possible_release_files = (
-        ReleaseFile.objects.filter(
-            organization_id=project.organization_id,
-            release_id=release.id,
-            name__in=filenme_choices,
-        )
-        .exclude(artifact_count=0)
-        .select_related("file")
-    )
-    if len(possible_release_files) > 0:
-        path_data["source_file_lookup_result"] = "wrong-dist"
-    for possible_release_file in possible_release_files:
+    artifact_index_release_files = ReleaseFile.objects.filter(
+        organization_id=project.organization_id,
+        release_id=release.id,
+        file__type="release.artifact-index",
+    ).select_related("file")[
+        :1000
+    ]  # limit by something sane in case people have an insane amount of uploads
```

Comment:
m: Do we have metrics that track p95 for release files per release? We can better decide on a magic number if we do.

We should also extract magic number into a constant.

Reply:
Ok, so I talked to @Swatinem and we have to pick a low-ish number here. Each element here will translate into a file download on the endpoint so I picked 25 for now. We can monitor the performance of this endpoint until we LA/GA its feature flag. I don't want to pick something that is _very_ likely to run into issues (i.e. 5) but let's see.

If we don't find it in the 25 results it's not too bad - we will just run into the `unsuccessful` case then and just show "we can't find the source map" in the debugger.